src/fides/api/models/privacy_notice.py

```diff
@@ -332,123 +365,249 @@ def create(
-        data.pop("id", None)
-        history_data = {**data, "privacy_notice_id": created.id}
-        PrivacyNoticeHistory.create(db, data=history_data, check_name=False)
         return created
 
     def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyNotice:
         """
-        Overrides the base update method to automatically bump the version of the
-        PrivacyNotice record and also create a new PrivacyNoticeHistory entry
+        Updates the Privacy Notice and its translations.
+
+        - Upserts or deletes supplied translations to match translations in the request
+        - For each remaining translation, create a historical record if the base notice
+        or translation changed.
         """
-        resource, updated = update_if_modified(self, db=db, data=data)
+        request_translations = data.pop("translations", [])
+
+        # Performs a patch update of the base privacy notice
+        base_notice_updated: bool = update_if_modified(self, db=db, data=data)
+
+        for translation_data in request_translations:
+            existing_translation: Optional[
+                NoticeTranslation
+            ] = self.get_translation_by_language(db, translation_data.get("language"))
+
+            if existing_translation:
+                translation: NoticeTranslation = existing_translation
+                # Performs a patch update of the existing translation
+                translation_updated: bool = update_if_modified(  # type: ignore[attr-defined]
+                    existing_translation,
+                    db=db,
+                    data={**translation_data, "privacy_notice_id": self.id},
+                )
+            else:
+                translation_updated = True
+                # Creates a new translation
+                translation = NoticeTranslation.create(
+                    db,
+                    data={**translation_data, "privacy_notice_id": self.id},
+                )
+
+            if base_notice_updated or translation_updated:
+                existing_version: float = translation.version or 0.0
+                history_data: dict = create_historical_data_from_record(self)
+                history_data.pop("translations", None)
+                updated_translation_data: dict = create_historical_data_from_record(
+                    translation
+                )
+                # Creates a historical record of the Notice and the translation combined. Preferences are saved
+                # against this resource.
+                PrivacyNoticeHistory.create(
+                    db,
+                    data={
+                        **history_data,
+                        **updated_translation_data,
+                        "translation_id": translation.id,
+                        "version": existing_version + 1.0,
+                    },
+                    check_name=False,
+                )
```

Comment:
again, i think there's probably some room for extracting these out into helpers, just to make the top-level method a bit cleaner. but not a major issue as-is!

Reply:
Two methods added like you requested above, one for the historical record, one for deleting the notice translations.  They could be further consolidated with the experience-level methods but I am thinking of keeping them separate mainly for type hinting purposes.